src/Modules/OrchardCore.Commerce/Services/PriceVariantProvider.cs

```diff
@@ -94,4 +94,21 @@ private ShoppingCartItem AddPriceToShoppingCartItem(ShoppingCartItem item, Produ
 
         return null;
     }
+
+    public async Task<bool> IsApplicableAsync(IList<ShoppingCartItem> items)
+    {
+        var skuProducts = await PriceProviderHelpers.GetSkuProductsAsync(items, _productService);
+
+        return items
+            .All(item =>
+            {
+                if (skuProducts.TryGetValue(item.ProductSku, out var productPart))
+                {
+                    var contentItem = productPart.ContentItem;
+                    return contentItem.OfType<PricePart>().Any() || contentItem.OfType<PriceVariantsPart>().Any();
+                }
+
+                return false;
+            });
```

Comment:
This can be shortened along [these lines](https://github.com/OrchardCMS/OrchardCore.Commerce/pull/146/files#r941884894), too.